ppml/README.md

```diff
@@ -0,0 +1,517 @@
+      
+   val ppmlArgs: Map[String, String] = Map(
+          "spark.bigdl.kms.type" -> "EHSMKeyManagementService",
+          "spark.bigdl.kms.ehs.ip" -> your_server_ip,
+          "spark.bigdl.kms.ehs.port" -> your_server_port,
+          "spark.bigdl.kms.ehs.id" -> your_app_id,
+          "spark.bigdl.kms.ehs.key" -> your_app_key,
+          "spark.bigdl.kms.key.primary" -> /your/primary/key/path,
+          "spark.bigdl.kms.key.data" -> /your/data/key/path
+   )
+      
+   val sc = PPMLContext.initPPMLContext("MyApp", ppmlArgs)
+   ```
+   
+   For `AzureKeyManagementService`
+   
+   ```scala
+   import com.intel.analytics.bigdl.ppml.PPMLContext
+      
+   val ppmlArgs: Map[String, String] = Map(
+          "spark.bigdl.kms.type" -> "AzureKeyManagementService",
+          "spark.bigdl.kms.azure.vault" -> key_vault_name,
+          "spark.bigdl.kms.azure.clientId" -> client_id,
+          "spark.bigdl.kms.key.primary" -> /your/primary/key/path,
+          "spark.bigdl.kms.key.data" -> /your/data/key/path
+      )
+      
+   val sc = PPMLContext.initPPMLContext("MyApp", ppmlArgs)
+   ```
+
+3. create a PPMLContext with `sparkConf` & `appName` & `ppmlArgs`
+
+   If you need to set Spark configurations, you can provide a `SparkConf` with Spark configurations to create a `PPMLContext`.
+
+   ```scala
+   import com.intel.analytics.bigdl.ppml.PPMLContext
+   import org.apache.spark.SparkConf
+   
+   val ppmlArgs: Map[String, String] = Map(
+       "spark.bigdl.kms.type" -> "SimpleKeyManagementService",
+       "spark.bigdl.kms.simple.id" -> your_app_id,
+       "spark.bigdl.kms.simple.key" -> your_app_key,
+       "spark.bigdl.kms.key.primary" -> /your/primary/key/path,
+       "spark.bigdl.kms.key.data" -> /your/data/key/path
+   )
+   
+   val conf: SparkConf = new SparkConf().setMaster("local[4]")
+   
+   val sc = PPMLContext.initPPMLContext(conf, "MyApp", ppmlArgs)
+   ```
+
+#### 2. Read & Write Files
+
+To read/write data, you should set the `CryptoMode`:
+
+- `plain_text`: no encryption
+- `AES/CBC/PKCS5Padding`: for CSV, JSON and text file
+- `AES_GCM_V1`: for PARQUET only
+- `AES_GCM_CTR_V1`: for PARQUET only
+
+To write data, you should set the `write` mode:
+
+- `overwrite`: Overwrite existing data with the content of dataframe.
+- `append`: Append content of the dataframe to existing data or table.
+- `ignore`: Ignore current write operation if data / table already exists without any error.
+- `error`: Throw an exception if data or table already exists.
+- `errorifexists`: Throw an exception if data or table already exists.
+
+<details><summary>expand to see the example of CryptoMode and write mode</summary>
+
+```scala
+import com.intel.analytics.bigdl.ppml.crypto.{AES_CBC_PKCS5PADDING, PLAIN_TEXT}
+
+// read data
+df = sc.read(cryptoMode = PLAIN_TEXT)
+	...
+
+// write data
+sc.write(dataFrame = df, cryptoMode = AES_CBC_PKCS5PADDING)
+.mode("overwrite")
+...
+```
+
+</details>
+
```

Comment:
<details><summary>expand to see the examples of reading/writing CSV, PARQUET, JSON and text file</summary>

In the following examples `sc` represents an initialized `PPMLContext`.

read/write CSV file
```
code
```

read/write PARQUET file
```
code
```

...